proposal/custom-url-resolution.md

```diff
@@ -0,0 +1,117 @@
+# Custom url resolution: Draft 1
+
+_[(Issue)](https://github.com/sass/sass/issues/2535)_
+
+## Table of Contents
+
+- [Background](#background)
+- [Summary](#summary)
+  - [Steps](#steps)
+  - [JavaScript API](#javaScript-api)
+  - [Edge cases](#edge-cases)
+- [Syntax](#syntax)
+- [Deprecation Process](#deprecation-process)
+
+## Background
+
+> This section is non-normative.
+
+Many css features require the use of a url import to reference resources from outside the sass files, however these files also need to exist on the eventual output directory and server. To ensure the references are valid, the sass API should allow for the user to provide a way to remap and/or inline these resources.
+
+## Summary
+
+> This section is non-normative.
+
+This proposal defines a standardized way to remap the url imports to the final location on the server or inline reference.
+
+This is accomplished by running url references through the importer plugin(s) and url rewriting plugin(s) if any url rewriting plugin has been defined.
+
+### Steps
+
+The steps of a url reference import:
```

Comment:
I'm not sure what "url reference import" means here. Do you mean plain-CSS `@import`s? Specifically only `@import`s that use the `url()` syntax for their URLs? Any `url()`?

A spec needs to be highly explicit, so there's no ambiguity and multiple implementations are able to implement exactly the same behavior. This is just the non-normative summary which has some more flexibility to be casual, but the expectation is that anything that's glossed over in the summary is made entirely explicit in the normative body of the proposal.

Reply:
I just meant url references. I think the wording should be less confusing now